Documentation/raspi-i2c.md

```diff
@@ -0,0 +1,162 @@
+
+```bash
+sudo nano /boot/config.txt
+```
+
+Add the line which will change the speed from 100_000 (default speed) to 10_000:
+
+```text
+dtparam=i2c_arm_baudrate=10000
+```
+
+Save the file with `ctrl + x` then `Y` then `enter`
+
+Then reboot:
+
+```bash
+sudo reboot
+```
+
+More information on dtoverlay and how to select specific elements of I2C buses are [available here](https://github.com/raspberrypi/firmware/blob/bff705fffe59ad3eea33999beb29c3f26408de40/boot/overlays/README#L1387). 2 busses are available on any Raspberry Pi. More can be available on specific models like those base out of BCM2711.
+
+### Enabling I2C with advance parameters
+
+The general pattern looks like the following:
+
+```text
+dtoverlay=i2cN,pins_A_B
+```
+
+Where:
+
+- N is the number os the I2C bus starting at 0
+- A is the SDA pin
+- B is the Clock pin
+
+Note:
+
+- i2c0 is equivalent to i2c_vc, do not use `dtoverlay` with `dtparam=i2c_vc=on`
+- i2c1 is equivalent to i2c_arm, this is the default one activated, do not use `dtoverlay` with `dtparam=i2c_arm=on`
+
+| I2C number | Authorized GPIO couple | dtoverlay |
+| --- | --- | --- |
+| I2C0 | GPIO 0 and 1 | dtoverlay=i2c0,pins_0_1 (default) |
+| I2C0 | GPIO 28 and 29 | dtoverlay=i2c0,pins_28_29 |
+| I2C0 | GPIO 44 and 45 | dtoverlay=i2c0,pins_44_45 |
+| I2C0 | GPIO 46 and 47 | dtoverlay=i2c0,pins_46_47 |
+| I2C1 | GPIO 2 and 3 | dtoverlay=i2c1,pins_2_3 (default) |
+| I2C1 | GPIO 44 and 45 | dtoverlay=i2c1,pins_44_45 |
+
+Following are only available on BCM2711. You can as well add `,baudrate=10000` for 10_000 or any other supported value to change the default baudrate which is 100_000:
+
+| I2C number | Authorized GPIO couple | dtoverlay |
+| --- | --- | --- |
+| I2C3 | GPIO 2 and 3 | dtoverlay=i2c3,pins_2_3 (default) |
+| I2C3 | GPIO 4 and 5 | dtoverlay=i2c3,pins_4_5 |
+| I2C4 | GPIO 6 and 7 | dtoverlay=i2c4,pins_6_7 |
+| I2C4 | GPIO 8 and 9 | dtoverlay=i2c4,pins_8_9 (default) |
+| I2C5 | GPIO 10 and 11 | dtoverlay=i2c5,pins_10_11 |
+| I2C5 | GPIO 12 and 13 | dtoverlay=i2c5,pins_12_13 (default) |
+| I2C6 | GPIO 0 and 1 | dtoverlay=i2c6,pins_0_1 |
+| I2C6 | GPIO 22 and 23 | dtoverlay=i2c6,pins_22_23 (default) |
+
+### Adding your user to the right permission group
+
+If you're running, or just upgraded to a version published after August 2020, this should be already done. 
+But in case, you can always check that there are the right permissions on I2C:
+
+```bash
+sudo nano /etc/udev/rules.d/99-com.rules
+```
+
+You should find a line like this one:
+
+```text
+SUBSYSTEM=="i2c-dev", GROUP="i2c", MODE="0660"
+```
+
+If you don't have it or if you want to adjust the permissions, this is what you'll need to add/adjust, as always save through `ctrl + x` then `Y` then `enter` and then reboot.
+
+## Bonus: write your own I2C scanner
```

Comment:
Maybe add a hint that there's already an existing tool to do this?

Reply:
Yes, good point. It does not exist on all systems. But right, it does on some! Will add!